src/pages/events/index.js

```diff
@@ -7,13 +7,14 @@ import { serverSideTranslations } from "next-i18next/serverSideTranslations";
 // React imports
 import React, { useEffect, useState } from "react";
 
+import { auth } from "@/lib/useAuth";
```

Comment:
Great work @SufyanCS! To follow the principle of `DRY` you will need to use the auth context without caring of doing `onAuthStateChanged` because all of that is being handled by the context.

So remove all authentication code and import the user from the context ```js
import { useAuthContext } from '../../../context/AuthContext'
const { user }  = useAuthContext();
```